types/dynamodb-lock-client/index.d.ts

```diff
@@ -0,0 +1,47 @@
+// Type definitions for dynamodb-lock-client 0.7
+// Project: https://github.com/tristanls/dynamodb-lock-client#readme
+// Definitions by: RyoshiKayo <https://github.com/RyoshiKayo>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import { DynamoDB } from "aws-sdk";
+import { EventEmitter } from "events";
+
+export interface Lock extends EventEmitter {
+    release(callback: (error: Error) => void): void;
+    fencingToken: number;
+}
+
+interface GenericConfig {
+    dynamodb: DynamoDB.DocumentClient;
+    lockTable: string;
+    partitionKey: string;
+    sortKey?: string;
+    owner?: string;
+}
+
+export interface FailClosedConfig extends GenericConfig {
+    acquirePeriodMs: number;
+}
+
+export interface FailOpenConfig extends GenericConfig {
+    heartbeatPeriodMs?: number;
+    leaseDurationMs: number;
+    trustLocalTime?: boolean;
+}
+
+export class LockClient<PartitionTableKeyType> {
+    acquireLock(id: PartitionTableKeyType, callback: (error: Error, lock: Lock) => void): void;
```

Comment:
Can `PartitionTableKeyType` be any type? The default of `string | Buffer | number` in the subclasses makes me wonder if it should be constrained to that type?

Reply:
Whoops I missed that, will fix.

I'm actually going to remove `Buffer` as well, I don't see any specific logic for buffers in the lib, and I don't see any repos (at least on Github) that are using Buffers as the ID.

Reply:
I think what you’re looking for is `extends string | number`, not `= string | number`. Or, you might be looking for both. But probably you don’t want _only_ the `=`. `extends` says that the key type should _always_ be a subtype of `string | number`. The `=` simply provides a default. So with this, you can still say `new LockClient(someBuffer, callback)` and get a `LockClient<Buffer>`. You’d need `extends string | number` to make that an error.

Reply:
Ah okay, that makes sense. Thank you for the explanation!

I also read up more on constraints: https://www.typescriptlang.org/docs/handbook/2/generics.html#generic-constraints